src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -61,10 +61,8 @@ public function boot()
     {
         ErrorHandler::register(null, false)->throwAt($this->container->getParameter('debug.error_handler.throw_at'), true);
 
-        if ($trustedProxies = $this->container->getParameter('kernel.trusted_proxies')) {
-            @trigger_error('The "kernel.trusted_proxies" parameter is deprecated since version 3.3 and will be removed in 4.0. Use the Request::setTrustedProxies() method in your front controller instead.', E_USER_DEPRECATED);
-
-            Request::setTrustedProxies($trustedProxies, Request::getTrustedHeaderSet());
+        if ($this->container->hasParameter('kernel.trusted_proxies')) {
```

Comment:
this will break the ``cache:clear`` command in non-debug environment when upgrading, as the existing cache (used when booting the kernel) will have this parameter defined.